### PR TITLE
Better handling of builtin CV attributes

### PR DESCRIPTION
**Pull request by leonerd**
The previous code would handle subroutine attributes directly against `PL_compcv` as a side-effect of merely parsing the syntax in `yyl_colon()`, an unlikely place for anyone to find it. This complicates the way the parser works.

The new structure creates a new function to apply all the builtin attributes out of an attribute list to any given CV, and invokes it from the parser at a slightly better time.

### CODE REVIEW

**Comment by leonerd** [comment]
The only failing test was `dist/Time-HiRes/t/alarm.t` on one of the MacOSX boxes; which I'm going to put down to being flaky and unrelated here.

File: perly.y [hunk 1]
```diff
@@ -709,7 +709,11 @@ proto
 subattrlist
 	:	empty
 	|	COLONATTR THING
-			{ $$ = $THING; }
+			{
+			  if($THING && !PL_parser->sig_seen)
+			      $THING = apply_builtin_cv_attributes(PL_compcv, $THING);
```
Comment (happy-barney) [inline perly.y]: better assign directly to `$$` than to named reference

Comment (leonerd) [reply perly.y]: Oh, fair. Actually probably better yet is to make a local variable out of it;

```C
OP *attrlist = $THING;
...
```

Comment (leonerd) [reply perly.y]: Adjusted and force-pushed. PTAL

File: toke.c [hunk 2]
```diff
@@ -5989,10 +5963,8 @@ yyl_colon(pTHX_ char *s)
                    rejected) by a package's attribute routines, but is
                    justified by the performance win for the common case
                    of applying only built-in attributes.) */
-                else
-                    attrs = op_append_elem(OP_LIST, attrs,
```
Comment (bram-perl) [inline toke.c:5993]: The comment above this block no longer make sense and should be updated.

Comment (leonerd) [reply toke.c:5993]: Now removed by d88cfc6975.

Needs squash before merge.

**Review by tonycoz — APPROVED** [review]

**Pull request merged**